lib/utils/index.js

```diff
@@ -83,9 +83,15 @@ async function credentialedDecodeResponse(response) {
     const decoded = await validateResponseAsJSON(response)
     return decoded
   } catch (e) {
-    // If we received unauthorized, map it to an invalid credentials error.
-    if (e instanceof errors.general.RequestError && e.response.status === 401) {
-      throw new errors.identity.InvalidCredentials(e.response, e.message)
+    if (e instanceof errors.general.RequestError) {
+      if (e.response.status === 401) {
+      // If we received unauthorized, map it to an invalid credentials error.
+        throw new errors.identity.InvalidCredentials(e.response, e.message)
+      }
+      if (e.response.status === 417) {
+      // If we received expectation failed, map it to an identity locked error.
+        throw new errors.identity.IdentityLockedError(e.response, e.message)
+      }
```

Comment:
for readability & a future world where we handle 418s or other error codes, i'd consider consolidating these `if` blocks into a `switch-case`
```js
switch (e.response.status) {
  case 401: // ...
  case 417: // ...
  default: break
}
```

![](https://http.cat/418)